### ISSUE TITLE
deribit - fix cost calculation

### ISSUE THREAD

**Issue by yrossslav** [ccxt/ccxt#7758]
Hi! Maybe I do break rules of issue submit. Sorry for that.

We may read [on deribit about amount](https://docs.deribit.com/#private-get_user_trades_by_currency): 

>Trade amount. For perpetual and futures - in USD units, for options it is amount of corresponding cryptocurrency contracts, e.g., BTC or ETH.

Bug in cost appear in parse_trade funciton, when you just calc: 

```
cost = amount * price
```

Not seeng what instrument_name is. It's return us problems with big cost like that: 

```
  'symbol': 'BTC-PERPETUAL',
  'type': 'limit',
  'side': 'buy',
  'takerOrMaker': 'maker',
  'price': 10608.0,
  'amount': 11830.0,
  'cost': 125492640.0,
```

**Comment by kroitor** [comment]
@yrossslav the unification of derivatives (perpetuals and futures) is still in progress, there may be quirks here and there, and we may end up redefining the meaning of `amount`, `price` and `cost` for the derivatives involving the settlement currency.
In this case, the terms used by deribit are confusing, since if the amount is given in USD, it's not the amount (not the number of contracts), it's the cost. We will investigate and will get back to you asap.

**Comment by yrossslav** [comment]
Yes! Thank you! I understand that you have a lot of work. Jus little help with bug report. I think that I found same little problems with bitmex, but, if it is in progress, I don't disturb you with that.

**Comment by ttodua** [comment]
> Yes! Thank you! I understand that you have a lot of work. Jus little help with bug report. I think that I found same little problems with bitmex, but, if it is in progress, I don't disturb you with that.

@yrossslav whenever you find any bug or issue (independent whether something is work-in-progress or not), please report. it's more then welcomed

**Comment by carlosmiei** [comment]
@m-nez Thanks for reporting it, we will take a look asap and get back to you

**Comment by m-nez** [comment]
The calculation of trade cost now works correctly for deribit spot markets with ccxt 4.4.30

**Comment by ttodua** [comment]
closing old topic, as it was resolved.
if anyone experiences any issues, let us know
